src/transformers/quantizers/quantizer_bitnet.py

```
@@ -0,0 +1,114 @@
+logger = logging.get_logger(__name__)
+
+
+class BitNetHfQuantizer(HfQuantizer):
+    """
+    1.58-bit quantization from BitNet quantization method:
+    before loading: converts transformer layers into BitLinear during loading.
+    """
+
+    requires_parameters_quantization = False
+    requires_calibration = True
+
+    required_packages = ["accelerate"]
+
+    def __init__(self, quantization_config, **kwargs):
+        super().__init__(quantization_config, **kwargs)
+        self.quantization_config = quantization_config
+
+    def validate_environment(self, *args, **kwargs):
+        if not is_accelerate_available():
+            raise ImportError("Loading a BitNet quantized model requires accelerate (`pip install accelerate`)")
+
+        if kwargs.get("from_tf", False) or kwargs.get("from_flax", False):
+            raise ValueError(
+                "Loading ternary weights from tf/flax is currently not supported, please make"
+                " sure the weights are in PyTorch format."
+            )
+
+        if not torch.cuda.is_available():
+            logger.warning_once(
+                "You don't have a GPU available to load the model, the inference will be slow because of weight unpacking"
+            )
+            return
+
+        device_map = kwargs.get("device_map", None)
+        if device_map is None:
+            logger.warning_once(
+                "You have loaded a BitNet model on CPU and have a CUDA device available, make sure to set "
+                "your model on a GPU device in order to run your model."
+            )
+        elif device_map is not None:
+            if isinstance(device_map, dict) and ("cpu" in device_map.values() or "disk" in device_map.values()):
+                logger.warning_once(
+                    "You are attempting to load a BitNet model with a device_map that contains a CPU or disk device."
+                    "This will degrade the inference speed because of weight unpacking"
+                )
```

Does it really work with cpu or disk offload ? Since the BitNetLinear is composed of buffers essentially, make sure to use with init_empty_weights(include_buffers=True) in `_replace_with_bitnet_linear` + device_map_kwargs["offload_buffers"] = True in modeling_utils.py. Check fbgemm-fp8 code. Otherwise, it won't offload the buffers correctly. 
We can do that in a follow up PR if needed and we can just raise an error here for now.